### DISCUSSION TITLE
Compilation Error in Arduino IDE

### DISCUSSION THREAD

**Discussion by danielerpl** [DIY-Flow-Bench/DIY-Flow-Bench#119] [Support]
Hi, I am trying to compile the code with Arduino IDE, but I am receiving this error `DIY-Flow-Bench/ESP32/DIY-Flow-Bench/calibration.cpp:19:0:
/Users/daniele/Documents/DIY-Flow-Bench/ESP32/DIY-Flow-Bench/calibration.h:36:3: error: 'StaticJsonDocument' does not name a type`

I tried to update everything but there is no solution to have the code up and running. 

I also tried to compile the code in visual studio but I have the same error in the StaticJsonDocument and another one for 
`[{
	"resource": "/Users/daniele/Documents/DIY-Flow-Bench/.pio/libdeps/release/Adafruit BusIO/Adafruit_SPIDevice.cpp",
	"owner": "cpp",
	"severity": 8,
	"message": "invalid conversion from 'const uint8_t* {aka const unsigned char*}' to 'uint8_t* {aka unsigned char*}' [-fpermissive]",
	"startLineNumber": 352,
	"startColumn": 61,
	"endLineNumber": 352,
	"endColumn": 61
}]`

Which IDE do you prefer ? Arduino IDE or Visual Studio ? 

Many thanks

**Comment by danielerpl** [comment]
Solved the problem for 'StaticJsonDocument' in Visual Studio Code. But I can't figure out which is the problem with `.pio/libdeps/release/Adafruit BusIO/Adafruit_SPIDevice.h:9:17: fatal error: SPI.h: No such file or directory`

**Comment by DeeEmm** [comment]
Hi onzio88r VSCode is definitely the preferred build environment. Whilst the code should compile on Arduino, it is many years since I have used the Arduino editor and so my knowledge for this is not so great. Generally the issues with Arduino relate to environment setup, such as missing or incorrectly installed libraries. Are you able to share how you resolved the initial issue? This will help others.

The fatal error: **fatal error: SPI.h: No such file or directory** is most likely also an environment issue. The message is showing that the library dependency Adafruit BusIO is either missing or not installed correctly.

This appears to be very strange as I do not use this library, nor use SPI. I2C is taken care of by the wire library

The following libraries are listed in the lib_deps section of platformio.ini that is supplied with the project:

```
lib_deps = 
	bblanchon/ArduinoJson@^6.19.4
	esphome/AsyncTCP-esphome
        esphome/ESPAsyncWebServer-esphome
	https://github.com/terryjmyers/ADS1115-Lite.git
	https://github.com/fabyte/Tiny_BME280_Arduino_Library.git
```

Generally all dependencies should be automatically resolved in VSCode as it uses the supplied platformio.ini file to determine which library files to pull in. 

Have you used the supplied platformio.ini file? Normally all that is required is to point VSCode at the root folder of the project (the folder that contains platformio.ini). This is not the same as Arduino IDE where you point the IDE at the DIY-Flow-bench sub folder.

/DM

**Reply by DeeEmm** [reply]
Also. Can you confirm the build number. You can find this in configuration.h

**Comment by DeeEmm** [comment]
How did you get on??

There is a new release which I advise you to try as it addresses a number of bugs

https://github.com/DeeEmm/DIY-Flow-Bench/releases/tag/V.2.0-RC.4

If you find any issues or require support with this version then please feel free to create another thread or comment in the official RC4 discussion

https://github.com/DeeEmm/DIY-Flow-Bench/discussions/121
